chainer/optimizers/adam.py

```diff
@@ -148,41 +151,48 @@ def update_core_gpu(self, param):
         if hp.amsgrad:
             if AdamRule._amsgrad_kernel is None:
                 AdamRule._amsgrad_kernel = cuda.elementwise(
-                    'T grad, T lr, T one_minus_beta1, T one_minus_beta2, '
+                    'T grad, T alpha_t, T one_minus_beta1, T one_minus_beta2, '
                     'T eps, T eta, T weight_decay_rate',
                     'T param, T m, T v, T vhat',
                     '''m += one_minus_beta1 * (grad - m);
                        v += one_minus_beta2 * (grad * grad - v);
                        vhat = max(vhat, v);
-                       param -= eta * (lr * m / (sqrt(vhat) + eps) +
+                       param -= eta * (alpha_t * m / (sqrt(vhat) + eps) +
                                        weight_decay_rate * param);''',
                     'adam')
             AdamRule._amsgrad_kernel(
-                grad, self.lr, 1 - hp.beta1,
+                grad, self.alpha_t, 1 - hp.beta1,
                 1 - hp.beta2, hp.eps,
                 hp.eta, hp.weight_decay_rate,
                 param.data, self.state['m'], self.state['v'],
                 self.state['vhat'])
         else:
             if AdamRule._kernel is None:
                 AdamRule._kernel = cuda.elementwise(
-                    'T grad, T lr, T one_minus_beta1, T one_minus_beta2, '
+                    'T grad, T alpha_t, T one_minus_beta1, T one_minus_beta2, '
                     'T eps, T eta, T weight_decay_rate',
                     'T param, T m, T v',
                     '''m += one_minus_beta1 * (grad - m);
                        v += one_minus_beta2 * (grad * grad - v);
-                       param -= eta * (lr * m / (sqrt(v) + eps) +
+                       param -= eta * (alpha_t * m / (sqrt(v) + eps) +
                                        weight_decay_rate * param);''',
                     'adam')
-            AdamRule._kernel(grad, self.lr, 1 - hp.beta1,
+            AdamRule._kernel(grad, self.alpha_t, 1 - hp.beta1,
                              1 - hp.beta2, hp.eps,
                              hp.eta, hp.weight_decay_rate,
                              param.data, self.state['m'], self.state['v'])
 
     @property
-    def lr(self):
+    def alpha_t(self):
         return _learning_rate(self.hyperparam, self.t)
 
+    @property
+    def lr(self):
+        warnings.warn(
+            'AdamRule.lr is deprecated. Use AdamRule.alpha_t instead.',
```

Comment:
I think it's better to improve the message so that users can clearly understand this is just a rename.
e.g., `AdamRule.lr has been renamed to AdamRule.alpha_t. Use of AdamRule.lr is deprecated in Chainer v6.`